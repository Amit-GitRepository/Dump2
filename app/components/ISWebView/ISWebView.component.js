import PropTypes from 'prop-types';
import React, {Component} from 'react';
import styles from './ISWebView.style';
import {OverlaySpinner} from '../Shared';
import {View, WebView} from 'react-native';

// fix https://github.com/facebook/react-native/issues/10865
const patchPostMessageJsCode = `(${String(function () {
  var originalPostMessage = window.postMessage;
  var patchedPostMessage = function (message, targetOrigin, transfer) {
    originalPostMessage(message, targetOrigin, transfer);
  };
  patchedPostMessage.toString = function () {
    return String(Object.hasOwnProperty).replace('hasOwnProperty', 'postMessage');
  };
  window.postMessage = patchedPostMessage;
  // Execute the javascript code now
  window.postMessage(document.title);
})})();`;

class ISWebView extends Component {

  state = {
    showSpinner: 1
  }

  setNavTitle = (titleText) => {
    const {navigation: {setParams}} = this.props;
    setParams({title: titleText});
  }

  onMessageReceived = (event) => {
    if (event.nativeEvent.data) {
      this.setNavTitle(event.nativeEvent.data);
    }
  }

  onLoadEnd = () => {
    this.setState({showSpinner: 0});
  }

  render () {
    const {url, headers} = this.props;
    const {showSpinner} = this.state;
    return (
      <View style={styles.container}>
        {url && <WebView
          source={
            {
              uri: url,
              headers
            }
          }
          injectedJavaScript={patchPostMessageJsCode}
          javaScriptEnabled={true}
          onMessage={this.onMessageReceived}
          onLoadEnd={this.onLoadEnd}
        />}
        {showSpinner > 0 && <OverlaySpinner showSpinner={showSpinner}/>}
      </View>
    );
  }
}

ISWebView.propTypes = {
  navigation: PropTypes.object,
  url: PropTypes.string.isRequired,
  headers: PropTypes.object
};

export default ISWebView;
