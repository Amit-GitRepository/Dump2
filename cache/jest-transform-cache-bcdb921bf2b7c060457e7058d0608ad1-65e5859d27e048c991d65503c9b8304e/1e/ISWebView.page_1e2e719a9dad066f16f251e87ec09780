Object.defineProperty(exports,"__esModule",{value:true});var cov_22f2tqcc50=function(){var path='/Users/amitkumar/iTrueService/DevCode/iservice-app/app/pages/ISWebView/ISWebView.page.js',hash='a58fc19c5b89e165c969a4dea5faaff6370526ca',global=new Function('return this')(),gcv='__coverage__',coverageData={path:'/Users/amitkumar/iTrueService/DevCode/iservice-app/app/pages/ISWebView/ISWebView.page.js',statementMap:{'0':{start:{line:8,column:20},end:{line:8,column:30}},'1':{start:{line:9,column:4},end:{line:11,column:6}},'2':{start:{line:15,column:25},end:{line:15,column:35}},'3':{start:{line:16,column:4},end:{line:18,column:6}},'4':{start:{line:22,column:0},end:{line:24,column:2}}},fnMap:{'0':{name:'(anonymous_0)',decl:{start:{line:7,column:29},end:{line:7,column:30}},loc:{start:{line:7,column:47},end:{line:12,column:3}},line:7},'1':{name:'(anonymous_1)',decl:{start:{line:14,column:2},end:{line:14,column:3}},loc:{start:{line:14,column:12},end:{line:19,column:3}},line:14}},branchMap:{},s:{'0':0,'1':0,'2':0,'3':0,'4':0},f:{'0':0,'1':0},b:{},_coverageSchema:'332fd63041d2c1bcb487cc26dd0d5f7d97098a6c'},coverage=global[gcv]||(global[gcv]={});if(coverage[path]&&coverage[path].hash===hash){return coverage[path];}coverageData.hash=hash;return coverage[path]=coverageData;}();var _createClass=function(){function defineProperties(target,props){for(var i=0;i<props.length;i++){var descriptor=props[i];descriptor.enumerable=descriptor.enumerable||false;descriptor.configurable=true;if("value"in descriptor)descriptor.writable=true;Object.defineProperty(target,descriptor.key,descriptor);}}return function(Constructor,protoProps,staticProps){if(protoProps)defineProperties(Constructor.prototype,protoProps);if(staticProps)defineProperties(Constructor,staticProps);return Constructor;};}();var _ISWebView=require('../../components/ISWebView/ISWebView.component');var _ISWebView2=_interopRequireDefault(_ISWebView);
var _propTypes=require('prop-types');var _propTypes2=_interopRequireDefault(_propTypes);
var _react=require('react');var _react2=_interopRequireDefault(_react);function _interopRequireDefault(obj){return obj&&obj.__esModule?obj:{default:obj};}function _classCallCheck(instance,Constructor){if(!(instance instanceof Constructor)){throw new TypeError("Cannot call a class as a function");}}function _possibleConstructorReturn(self,call){if(!self){throw new ReferenceError("this hasn't been initialised - super() hasn't been called");}return call&&(typeof call==="object"||typeof call==="function")?call:self;}function _inherits(subClass,superClass){if(typeof superClass!=="function"&&superClass!==null){throw new TypeError("Super expression must either be null or a function, not "+typeof superClass);}subClass.prototype=Object.create(superClass&&superClass.prototype,{constructor:{value:subClass,enumerable:false,writable:true,configurable:true}});if(superClass)Object.setPrototypeOf?Object.setPrototypeOf(subClass,superClass):subClass.__proto__=superClass;}var

ISWebViewPage=function(_Component){_inherits(ISWebViewPage,_Component);function ISWebViewPage(){_classCallCheck(this,ISWebViewPage);return _possibleConstructorReturn(this,(ISWebViewPage.__proto__||Object.getPrototypeOf(ISWebViewPage)).apply(this,arguments));}_createClass(ISWebViewPage,[{key:'render',value:function render()








{cov_22f2tqcc50.f[1]++;var _ref=(cov_22f2tqcc50.s[2]++,
this.props),navigation=_ref.navigation;cov_22f2tqcc50.s[3]++;
return(
_react2.default.createElement(_ISWebView2.default,{navigation:navigation,headers:navigation.state.params.headers,url:navigation.state.params.url}));

}}]);return ISWebViewPage;}(_react.Component);ISWebViewPage.navigationOptions=function(_ref2){var navigation=_ref2.navigation;cov_22f2tqcc50.f[0]++;var _ref3=(cov_22f2tqcc50.s[0]++,navigation),state=_ref3.state;cov_22f2tqcc50.s[1]++;return{title:state.params.title};};cov_22f2tqcc50.s[4]++;


ISWebViewPage.propTypes={
navigation:_propTypes2.default.object};exports.default=


ISWebViewPage;