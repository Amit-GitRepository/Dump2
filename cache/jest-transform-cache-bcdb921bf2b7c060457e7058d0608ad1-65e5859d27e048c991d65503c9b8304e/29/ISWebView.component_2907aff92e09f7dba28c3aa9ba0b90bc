Object.defineProperty(exports,"__esModule",{value:true});var cov_1he908k5cg=function(){var path='/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/ISWebView/ISWebView.component.js',hash='0a0c28b9175d2984679d9daa10ea7090e7f61efe',global=new Function('return this')(),gcv='__coverage__',coverageData={path:'/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/ISWebView/ISWebView.component.js',statementMap:{'0':{start:{line:7,column:31},end:{line:18,column:8}},'1':{start:{line:8,column:28},end:{line:8,column:46}},'2':{start:{line:9,column:27},end:{line:11,column:3}},'3':{start:{line:10,column:4},end:{line:10,column:57}},'4':{start:{line:12,column:2},end:{line:14,column:4}},'5':{start:{line:13,column:4},end:{line:13,column:82}},'6':{start:{line:15,column:2},end:{line:15,column:42}},'7':{start:{line:17,column:2},end:{line:17,column:37}},'8':{start:{line:23,column:38},end:{line:23,column:48}},'9':{start:{line:24,column:4},end:{line:24,column:34}},'10':{start:{line:28,column:4},end:{line:30,column:5}},'11':{start:{line:29,column:6},end:{line:29,column:47}},'12':{start:{line:34,column:18},end:{line:34,column:28}},'13':{start:{line:35,column:4},end:{line:45,column:6}},'14':{start:{line:49,column:0},end:{line:52,column:2}}},fnMap:{'0':{name:'(anonymous_0)',decl:{start:{line:7,column:42},end:{line:7,column:43}},loc:{start:{line:7,column:54},end:{line:18,column:1}},line:7},'1':{name:'(anonymous_1)',decl:{start:{line:9,column:27},end:{line:9,column:28}},loc:{start:{line:9,column:70},end:{line:11,column:3}},line:9},'2':{name:'(anonymous_2)',decl:{start:{line:12,column:32},end:{line:12,column:33}},loc:{start:{line:12,column:44},end:{line:14,column:3}},line:12},'3':{name:'(anonymous_3)',decl:{start:{line:22,column:16},end:{line:22,column:17}},loc:{start:{line:22,column:31},end:{line:25,column:3}},line:22},'4':{name:'(anonymous_4)',decl:{start:{line:27,column:22},end:{line:27,column:23}},loc:{start:{line:27,column:33},end:{line:31,column:3}},line:27},'5':{name:'(anonymous_5)',decl:{start:{line:33,column:2},end:{line:33,column:3}},loc:{start:{line:33,column:12},end:{line:46,column:3}},line:33}},branchMap:{'0':{loc:{start:{line:28,column:4},end:{line:30,column:5}},type:'if',locations:[{start:{line:28,column:4},end:{line:30,column:5}},{start:{line:28,column:4},end:{line:30,column:5}}],line:28},'1':{loc:{start:{line:37,column:9},end:{line:43,column:10}},type:'binary-expr',locations:[{start:{line:37,column:9},end:{line:37,column:12}},{start:{line:37,column:16},end:{line:43,column:10}}],line:37}},s:{'0':0,'1':0,'2':0,'3':0,'4':0,'5':0,'6':0,'7':0,'8':0,'9':0,'10':0,'11':0,'12':0,'13':0,'14':0},f:{'0':0,'1':0,'2':0,'3':0,'4':0,'5':0},b:{'0':[0,0],'1':[0,0]},_coverageSchema:'332fd63041d2c1bcb487cc26dd0d5f7d97098a6c'},coverage=global[gcv]||(global[gcv]={});if(coverage[path]&&coverage[path].hash===hash){return coverage[path];}coverageData.hash=hash;return coverage[path]=coverageData;}();var _createClass=function(){function defineProperties(target,props){for(var i=0;i<props.length;i++){var descriptor=props[i];descriptor.enumerable=descriptor.enumerable||false;descriptor.configurable=true;if("value"in descriptor)descriptor.writable=true;Object.defineProperty(target,descriptor.key,descriptor);}}return function(Constructor,protoProps,staticProps){if(protoProps)defineProperties(Constructor.prototype,protoProps);if(staticProps)defineProperties(Constructor,staticProps);return Constructor;};}();var _propTypes=require('prop-types');var _propTypes2=_interopRequireDefault(_propTypes);
var _react=require('react');var _react2=_interopRequireDefault(_react);
var _ISWebView=require('./ISWebView.style');var _ISWebView2=_interopRequireDefault(_ISWebView);
var _reactNative=require('react-native');function _interopRequireDefault(obj){return obj&&obj.__esModule?obj:{default:obj};}function _classCallCheck(instance,Constructor){if(!(instance instanceof Constructor)){throw new TypeError("Cannot call a class as a function");}}function _possibleConstructorReturn(self,call){if(!self){throw new ReferenceError("this hasn't been initialised - super() hasn't been called");}return call&&(typeof call==="object"||typeof call==="function")?call:self;}function _inherits(subClass,superClass){if(typeof superClass!=="function"&&superClass!==null){throw new TypeError("Super expression must either be null or a function, not "+typeof superClass);}subClass.prototype=Object.create(superClass&&superClass.prototype,{constructor:{value:subClass,enumerable:false,writable:true,configurable:true}});if(superClass)Object.setPrototypeOf?Object.setPrototypeOf(subClass,superClass):subClass.__proto__=superClass;}


var patchPostMessageJsCode=(cov_1he908k5cg.s[0]++,'('+String(function(){cov_1he908k5cg.f[0]++;
var originalPostMessage=(cov_1he908k5cg.s[1]++,window.postMessage);cov_1he908k5cg.s[2]++;
var patchedPostMessage=function patchedPostMessage(message,targetOrigin,transfer){cov_1he908k5cg.f[1]++;cov_1he908k5cg.s[3]++;
originalPostMessage(message,targetOrigin,transfer);
};cov_1he908k5cg.s[4]++;
patchedPostMessage.toString=function(){cov_1he908k5cg.f[2]++;cov_1he908k5cg.s[5]++;
return String(Object.hasOwnProperty).replace('hasOwnProperty','postMessage');
};cov_1he908k5cg.s[6]++;
window.postMessage=patchedPostMessage;cov_1he908k5cg.s[7]++;

window.postMessage(document.title);
})+')();');var

ISWebView=function(_Component){_inherits(ISWebView,_Component);function ISWebView(){var _ref;var _temp,_this,_ret;_classCallCheck(this,ISWebView);for(var _len=arguments.length,args=Array(_len),_key=0;_key<_len;_key++){args[_key]=arguments[_key];}return _ret=(_temp=(_this=_possibleConstructorReturn(this,(_ref=ISWebView.__proto__||Object.getPrototypeOf(ISWebView)).call.apply(_ref,[this].concat(args))),_this),_this.

setNavTitle=function(titleText){cov_1he908k5cg.f[3]++;var _ref2=(cov_1he908k5cg.s[8]++,
_this.props),setParams=_ref2.navigation.setParams;cov_1he908k5cg.s[9]++;
setParams({title:titleText});
},_this.

onMessageReceived=function(event){cov_1he908k5cg.f[4]++;cov_1he908k5cg.s[10]++;
if(event.nativeEvent.data){cov_1he908k5cg.b[0][0]++;cov_1he908k5cg.s[11]++;
_this.setNavTitle(event.nativeEvent.data);
}else{cov_1he908k5cg.b[0][1]++;}
},_temp),_possibleConstructorReturn(_this,_ret);}_createClass(ISWebView,[{key:'render',value:function render()

{cov_1he908k5cg.f[5]++;var _ref3=(cov_1he908k5cg.s[12]++,
this.props),url=_ref3.url;cov_1he908k5cg.s[13]++;
return(
_react2.default.createElement(_reactNative.View,{style:_ISWebView2.default.container},
(cov_1he908k5cg.b[1][0]++,url)&&(cov_1he908k5cg.b[1][1]++,_react2.default.createElement(_reactNative.WebView,{
source:{uri:url},
injectedJavaScript:patchPostMessageJsCode,
startInLoadingState:true,
javaScriptEnabled:true,
onMessage:this.onMessageReceived}))));



}}]);return ISWebView;}(_react.Component);cov_1he908k5cg.s[14]++;


ISWebView.propTypes={
navigation:_propTypes2.default.object,
url:_propTypes2.default.string.isRequired};exports.default=


ISWebView;