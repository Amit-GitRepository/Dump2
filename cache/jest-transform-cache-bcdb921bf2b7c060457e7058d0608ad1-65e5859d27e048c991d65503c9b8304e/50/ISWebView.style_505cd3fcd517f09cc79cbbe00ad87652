Object.defineProperty(exports,"__esModule",{value:true});var cov_y3vxjkccw=function(){var path="/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/ISWebView/ISWebView.style.js",hash="9ccad57af722273c800ee334b545236016ec82c9",global=new Function('return this')(),gcv="__coverage__",coverageData={path:"/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/ISWebView/ISWebView.style.js",statementMap:{},fnMap:{},branchMap:{},s:{},f:{},b:{},_coverageSchema:"332fd63041d2c1bcb487cc26dd0d5f7d97098a6c"},coverage=global[gcv]||(global[gcv]={});if(coverage[path]&&coverage[path].hash===hash){return coverage[path];}coverageData.hash=hash;return coverage[path]=coverageData;}();exports.default={
container:{
flex:1}};