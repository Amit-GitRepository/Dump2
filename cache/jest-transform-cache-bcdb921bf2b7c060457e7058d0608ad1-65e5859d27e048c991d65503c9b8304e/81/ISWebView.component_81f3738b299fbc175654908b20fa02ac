var _ISWebView=require('../ISWebView.component');var _ISWebView2=_interopRequireDefault(_ISWebView);
var _react=require('react');var _react2=_interopRequireDefault(_react);
var _reactTestRenderer=require('react-test-renderer');var _reactTestRenderer2=_interopRequireDefault(_reactTestRenderer);function _interopRequireDefault(obj){return obj&&obj.__esModule?obj:{default:obj};}

describe('ISWebView component',function(){
it('renders correctly',function(){
var component=_reactTestRenderer2.default.create(_react2.default.createElement(_ISWebView2.default,null)).toJSON();
expect(component).toMatchSnapshot();
});
});