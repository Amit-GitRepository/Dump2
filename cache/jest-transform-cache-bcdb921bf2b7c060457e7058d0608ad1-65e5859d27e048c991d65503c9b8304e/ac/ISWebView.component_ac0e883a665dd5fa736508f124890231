Object.defineProperty(exports,"__esModule",{value:true});var cov_1he908k5cg=function(){var path='/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/ISWebView/ISWebView.component.js',hash='8d59a2789b882fdca33abb4961b758c56a1542d8',global=new Function('return this')(),gcv='__coverage__',coverageData={path:'/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/ISWebView/ISWebView.component.js',statementMap:{'0':{start:{line:8,column:31},end:{line:19,column:8}},'1':{start:{line:9,column:28},end:{line:9,column:46}},'2':{start:{line:10,column:27},end:{line:12,column:3}},'3':{start:{line:11,column:4},end:{line:11,column:57}},'4':{start:{line:13,column:2},end:{line:15,column:4}},'5':{start:{line:14,column:4},end:{line:14,column:82}},'6':{start:{line:16,column:2},end:{line:16,column:42}},'7':{start:{line:18,column:2},end:{line:18,column:37}},'8':{start:{line:28,column:38},end:{line:28,column:48}},'9':{start:{line:29,column:4},end:{line:29,column:34}},'10':{start:{line:33,column:4},end:{line:35,column:5}},'11':{start:{line:34,column:6},end:{line:34,column:47}},'12':{start:{line:39,column:4},end:{line:39,column:36}},'13':{start:{line:43,column:27},end:{line:43,column:37}},'14':{start:{line:44,column:26},end:{line:44,column:36}},'15':{start:{line:45,column:4},end:{line:61,column:6}},'16':{start:{line:65,column:0},end:{line:69,column:2}}},fnMap:{'0':{name:'(anonymous_0)',decl:{start:{line:8,column:42},end:{line:8,column:43}},loc:{start:{line:8,column:54},end:{line:19,column:1}},line:8},'1':{name:'(anonymous_1)',decl:{start:{line:10,column:27},end:{line:10,column:28}},loc:{start:{line:10,column:70},end:{line:12,column:3}},line:10},'2':{name:'(anonymous_2)',decl:{start:{line:13,column:32},end:{line:13,column:33}},loc:{start:{line:13,column:44},end:{line:15,column:3}},line:13},'3':{name:'(anonymous_3)',decl:{start:{line:27,column:16},end:{line:27,column:17}},loc:{start:{line:27,column:31},end:{line:30,column:3}},line:27},'4':{name:'(anonymous_4)',decl:{start:{line:32,column:22},end:{line:32,column:23}},loc:{start:{line:32,column:33},end:{line:36,column:3}},line:32},'5':{name:'(anonymous_5)',decl:{start:{line:38,column:14},end:{line:38,column:15}},loc:{start:{line:38,column:20},end:{line:40,column:3}},line:38},'6':{name:'(anonymous_6)',decl:{start:{line:42,column:2},end:{line:42,column:3}},loc:{start:{line:42,column:12},end:{line:62,column:3}},line:42}},branchMap:{'0':{loc:{start:{line:33,column:4},end:{line:35,column:5}},type:'if',locations:[{start:{line:33,column:4},end:{line:35,column:5}},{start:{line:33,column:4},end:{line:35,column:5}}],line:33},'1':{loc:{start:{line:47,column:9},end:{line:58,column:10}},type:'binary-expr',locations:[{start:{line:47,column:9},end:{line:47,column:12}},{start:{line:47,column:16},end:{line:58,column:10}}],line:47},'2':{loc:{start:{line:59,column:9},end:{line:59,column:71}},type:'binary-expr',locations:[{start:{line:59,column:9},end:{line:59,column:24}},{start:{line:59,column:28},end:{line:59,column:71}}],line:59}},s:{'0':0,'1':0,'2':0,'3':0,'4':0,'5':0,'6':0,'7':0,'8':0,'9':0,'10':0,'11':0,'12':0,'13':0,'14':0,'15':0,'16':0},f:{'0':0,'1':0,'2':0,'3':0,'4':0,'5':0,'6':0},b:{'0':[0,0],'1':[0,0],'2':[0,0]},_coverageSchema:'332fd63041d2c1bcb487cc26dd0d5f7d97098a6c'},coverage=global[gcv]||(global[gcv]={});if(coverage[path]&&coverage[path].hash===hash){return coverage[path];}coverageData.hash=hash;return coverage[path]=coverageData;}();var _createClass=function(){function defineProperties(target,props){for(var i=0;i<props.length;i++){var descriptor=props[i];descriptor.enumerable=descriptor.enumerable||false;descriptor.configurable=true;if("value"in descriptor)descriptor.writable=true;Object.defineProperty(target,descriptor.key,descriptor);}}return function(Constructor,protoProps,staticProps){if(protoProps)defineProperties(Constructor.prototype,protoProps);if(staticProps)defineProperties(Constructor,staticProps);return Constructor;};}();var _propTypes=require('prop-types');var _propTypes2=_interopRequireDefault(_propTypes);
var _react=require('react');var _react2=_interopRequireDefault(_react);
var _ISWebView=require('./ISWebView.style');var _ISWebView2=_interopRequireDefault(_ISWebView);
var _Shared=require('../Shared');
var _reactNative=require('react-native');function _interopRequireDefault(obj){return obj&&obj.__esModule?obj:{default:obj};}function _classCallCheck(instance,Constructor){if(!(instance instanceof Constructor)){throw new TypeError("Cannot call a class as a function");}}function _possibleConstructorReturn(self,call){if(!self){throw new ReferenceError("this hasn't been initialised - super() hasn't been called");}return call&&(typeof call==="object"||typeof call==="function")?call:self;}function _inherits(subClass,superClass){if(typeof superClass!=="function"&&superClass!==null){throw new TypeError("Super expression must either be null or a function, not "+typeof superClass);}subClass.prototype=Object.create(superClass&&superClass.prototype,{constructor:{value:subClass,enumerable:false,writable:true,configurable:true}});if(superClass)Object.setPrototypeOf?Object.setPrototypeOf(subClass,superClass):subClass.__proto__=superClass;}


var patchPostMessageJsCode=(cov_1he908k5cg.s[0]++,'('+String(function(){cov_1he908k5cg.f[0]++;
var originalPostMessage=(cov_1he908k5cg.s[1]++,window.postMessage);cov_1he908k5cg.s[2]++;
var patchedPostMessage=function patchedPostMessage(message,targetOrigin,transfer){cov_1he908k5cg.f[1]++;cov_1he908k5cg.s[3]++;
originalPostMessage(message,targetOrigin,transfer);
};cov_1he908k5cg.s[4]++;
patchedPostMessage.toString=function(){cov_1he908k5cg.f[2]++;cov_1he908k5cg.s[5]++;
return String(Object.hasOwnProperty).replace('hasOwnProperty','postMessage');
};cov_1he908k5cg.s[6]++;
window.postMessage=patchedPostMessage;cov_1he908k5cg.s[7]++;

window.postMessage(document.title);
})+')();');var

ISWebView=function(_Component){_inherits(ISWebView,_Component);function ISWebView(){var _ref;var _temp,_this,_ret;_classCallCheck(this,ISWebView);for(var _len=arguments.length,args=Array(_len),_key=0;_key<_len;_key++){args[_key]=arguments[_key];}return _ret=(_temp=(_this=_possibleConstructorReturn(this,(_ref=ISWebView.__proto__||Object.getPrototypeOf(ISWebView)).call.apply(_ref,[this].concat(args))),_this),_this.

state={
showSpinner:1},_this.


setNavTitle=function(titleText){cov_1he908k5cg.f[3]++;var _ref2=(cov_1he908k5cg.s[8]++,
_this.props),setParams=_ref2.navigation.setParams;cov_1he908k5cg.s[9]++;
setParams({title:titleText});
},_this.

onMessageReceived=function(event){cov_1he908k5cg.f[4]++;cov_1he908k5cg.s[10]++;
if(event.nativeEvent.data){cov_1he908k5cg.b[0][0]++;cov_1he908k5cg.s[11]++;
_this.setNavTitle(event.nativeEvent.data);
}else{cov_1he908k5cg.b[0][1]++;}
},_this.

onLoadEnd=function(){cov_1he908k5cg.f[5]++;cov_1he908k5cg.s[12]++;
_this.setState({showSpinner:0});
},_temp),_possibleConstructorReturn(_this,_ret);}_createClass(ISWebView,[{key:'render',value:function render()

{cov_1he908k5cg.f[6]++;var _ref3=(cov_1he908k5cg.s[13]++,
this.props),url=_ref3.url,headers=_ref3.headers;var _ref4=(cov_1he908k5cg.s[14]++,
this.state),showSpinner=_ref4.showSpinner;cov_1he908k5cg.s[15]++;
return(
_react2.default.createElement(_reactNative.View,{style:_ISWebView2.default.container},
(cov_1he908k5cg.b[1][0]++,url)&&(cov_1he908k5cg.b[1][1]++,_react2.default.createElement(_reactNative.WebView,{
source:
{
uri:url,
headers:headers},


injectedJavaScript:patchPostMessageJsCode,
javaScriptEnabled:true,
onMessage:this.onMessageReceived,
onLoadEnd:this.onLoadEnd})),

(cov_1he908k5cg.b[2][0]++,showSpinner>0)&&(cov_1he908k5cg.b[2][1]++,_react2.default.createElement(_Shared.OverlaySpinner,{showSpinner:showSpinner}))));


}}]);return ISWebView;}(_react.Component);cov_1he908k5cg.s[16]++;


ISWebView.propTypes={
navigation:_propTypes2.default.object,
url:_propTypes2.default.string.isRequired,
headers:_propTypes2.default.object};exports.default=


ISWebView;