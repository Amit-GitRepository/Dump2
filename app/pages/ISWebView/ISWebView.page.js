import ISWebView from '../../components/ISWebView/ISWebView.component';
import PropTypes from 'prop-types';
import React, {Component} from 'react';

class ISWebViewPage extends Component {

  static navigationOptions = ({navigation}) => {
    const {state} = navigation;
    return {
      title: state.params.title
    };
  };

  render () {
    const {navigation} = this.props;
    return (
      <ISWebView navigation={navigation} headers={navigation.state.params.headers} url={navigation.state.params.url} />
    );
  }
}

ISWebViewPage.propTypes = {
  navigation: PropTypes.object
};

export default ISWebViewPage;
