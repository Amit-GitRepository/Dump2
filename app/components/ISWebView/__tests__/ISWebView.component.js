import ISWebView from '../ISWebView.component';
import React from 'react';
import renderer from 'react-test-renderer'; // Enzyme adapter for React 16

describe('ISWebView component', () => {
  it('renders correctly', () => {
    const component = renderer.create(<ISWebView />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
