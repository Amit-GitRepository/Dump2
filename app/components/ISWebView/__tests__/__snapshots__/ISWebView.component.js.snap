// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ISWebView component renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "rgba(48,60,70,0.9)",
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      animation="zoomIn"
      duration={1000}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <ActivityIndicator
        animating={true}
        color="#FFFFFF"
        hidesWhenStopped={true}
        size="large"
      />
    </View>
  </View>
</View>
`;
